src/cascadia/PublicTerminalCore/HwndTerminal.cpp

```diff
@@ -141,15 +159,52 @@ void HwndTerminal::RegisterWriteCallback(const void _stdcall callback(wchar_t*))
     });
 }
 
+::Microsoft::Console::Types::IUiaData* HwndTerminal::GetUiaData() const
+{
+    return _terminal.get();
+}
+
+HWND HwndTerminal::GetHwnd() const
+{
+    return _hwnd.get();
+}
+
 void HwndTerminal::_UpdateFont(int newDpi)
 {
+    _currentDpi = newDpi;
     auto lock = _terminal->LockForWriting();
 
     // TODO: MSFT:20895307 If the font doesn't exist, this doesn't
     //      actually fail. We need a way to gracefully fallback.
     _renderer->TriggerFontChange(newDpi, _desiredFont, _actualFont);
 }
 
+IRawElementProviderSimple* HwndTerminal::_GetUiaProvider() noexcept
+{
+    if (nullptr == _uiaProvider && !_uiaProviderInitialized)
+    {
+        std::unique_lock<std::shared_mutex> lock;
+        try
+        {
+            lock = _terminal->LockForWriting();

```

Comment:
We generally use this for locks but I'm not sure if it applies here. So I'm just gonna write it down:
```
    _pData->LockConsole();
    auto Unlock = wil::scope_exit([&]() noexcept {
        _pData->UnlockConsole();
    });
```

Reply:
`LockConsole()` doesn't return a unique lock, so it needs the `scope_exit` closure to unlock it. the `LockForWriting()` method already returns a `std::unique_lock` which will unlock at the end of scope.